components/public-api/gitpod/v1/prebuild.proto

```diff
@@ -0,0 +1,107 @@
+syntax = "proto3";
+
+package gitpod.v1;
+
+import "google/protobuf/timestamp.proto";
+import "gitpod/v1/pagination.proto";
+
+option go_package = "github.com/gitpod-io/gitpod/components/public-api/go/v1";
+
+service PrebuildService {
+    rpc StartPrebuild(StartPrebuildRequest) returns (StartPrebuildResponse) {}
+    rpc StopPrebuild(StopPrebuildRequest) returns (StopPrebuildResponse) {}
+
+    rpc GetPrebuild(GetPrebuildRequest) returns (GetPrebuildResponse) {}
+    rpc ListPrebuilds(ListPrebuildsRequest) returns (ListPrebuildsResponse) {}
+    rpc WatchPrebuild(WatchPrebuildRequest) returns (stream WatchPrebuildResponse) {}
+}
+
+message GetPrebuildRequest {
+    string prebuild_id = 1;
+    string workspace_id = 2;
+    string configuration_id = 3;
+    string branch = 4;
+}
+message GetPrebuildResponse {
+    Prebuild prebuild = 1;
+}
+
+message ListPrebuildsRequest {
+    PaginationRequest pagination = 1;
+    string prebuild_id = 2;
+    string configuration_id = 3;
+    string branch = 4;
+}
+
+message ListPrebuildsResponse {
+    PaginationResponse pagination = 1;
+    repeated Prebuild prebuilds = 2;
+}
+
+message StartPrebuildRequest {
+    string configuration_id = 1;
+    string branch = 2;
+}
+message StartPrebuildResponse {
+    string prebuild_id = 1;
+}
+
+message StopPrebuildRequest {
+    string prebuild_id = 1;
+    // waht for?
+    string configuration_id = 2;
+}
+message StopPrebuildResponse {
+}
+
+message WatchPrebuildRequest {
+    string prebuild_id = 1;
+    string workspace_id = 2;
+    string configuration_id = 3;
+}
+message WatchPrebuildResponse {
+    Prebuild prebuild = 1;
+}
+
+message Author {
+    string name = 1;
+    string avatar_url = 2;
+}
+
+message Commit {
+    string message = 1;
+    Author author = 2;
+    google.protobuf.Timestamp author_date = 3;
+    string sha = 4;
+}
+
+message Prebuild {
+    string id = 1;
+    string based_on_prebuild_id = 2;
+    string build_workspace_id = 3;
```

Comment:
IMO we should try to keep this field internal (cmp. [comment](https://github.com/gitpod-io/gitpod/pull/19062/files#r1390766274) above), to the extend of marking in deprecated from the start maybe?

Reply:
I will add a comment, ok? Unfortunately the dashboard rely on it. I will double check places though whether we can do it differently. Generally everything in proto is used by the dashboard.